### ISSUE TITLE
Locus "insufficient coverage for plot" with modbam 

### ISSUE THREAD

**Issue by nmflack** [adamewing/methylartist#21]
Good afternoon,

I'm getting an error while running `methylartist locus` on a haplotagged modBAM generated with PEPPER-MARGIN-DEEPVARIANT. The BAM contains Mm and Ml tags and works fine with other programs like modbamtools and modbam2bed. The call and output are: 

```methylartist locus --phased --color_by_hp --ref no_ctg001250_curated.wrapped.fasta --motif CG -b pepper_modbam/mod.haplotagged.bam -i ctg001660:139662493-139666498```

```
2022-08-03 15:42:42,209 mod motif size (--motifsize) = 2 (ensure this is correct for your data)
2022-08-03 15:42:42,251 phases for bam pepper_modbam/mod.haplotagged.bam: 
2022-08-03 15:42:42,294 found mods: m in bam pepper_modbam/mod.haplotagged.bam
ctg001660:139662493-139666498: insufficient coverage for plot
```

Running `samtools coverage` for the region tells me that coverage should be sufficient:

```
samtools coverage -r ctg001660:139662493-139666498 mod.haplotagged.bam
#rname	startpos	endpos	numreads	covbases	coverage	meandepth	meanbaseq	meanmapq
ctg001660	139662493	139666498	80	4006	100	56.8635	20.4	60
```

I tried running `region` instead with the same flag values.

`methylartist region --phased --color_by_hp --ref no_ctg001250_curated.wrapped.fasta --motif CG -b pepper_modbam/mod.haplotagged.bam -i ctg001660:139662493-139666498`

It tries a little harder but gives this output: 

```
2022-08-03 15:46:36,550 locus smaller than 0.5 Mbp, "methylartist locus" may yield better results
2022-08-03 15:46:36,551 set window count to 21
2022-08-03 15:46:36,551 resetting windows to a minimum of 500
2022-08-03 15:46:36,551 motif count: 630, per window: 1
2022-08-03 15:46:36,554 using 316 windows normalised for CG content
2022-08-03 15:46:36,554 set --smoothwindowsize to 10
2022-08-03 15:46:36,554 set --modspace to 1
2022-08-03 15:46:36,608 found mods: m
2022-08-03 15:46:36,608 using mods m
2022-08-03 15:46:36,608 fetching reads from pepper_modbam/mod.haplotagged.bam with mod m on phase 1
2022-08-03 15:46:36,710 fetching reads from pepper_modbam/mod.haplotagged.bam with mod m on phase 2
2022-08-03 15:46:36,820 parsing segments (phase 1)...
2022-08-03 15:47:08,918 parsing segments (phase 2)...
2022-08-03 15:47:40,571 100.00 percent of windows for mod m had less than 1 calls
greater than 50.00 windows are uncovered, aborting.
```

Lastly, `wgmeth` is also not a fan of the modBAM. I made DSS input files manually with `modbam2bed` instead and successfully called DMRs, but I'd love to visualize them with methylartist. I only have that error as a screenshot:

<img width="1203" alt="ma" src="https://user-images.githubusercontent.com/109246157/182709979-d6eed92c-89f0-4066-9268-a1187c1ebacb.png">

I appreciate your input on what might be causing these errors. I'm running Ubuntu 20.04.4 LTS and methylartist version 1.2.3 installed in a clean conda environment.

Thank you,
Nicole

**Comment by adamewing** [comment]
Hi Nicole, is there any possibility you could share `pepper_modbam/mod.haplotagged.bam` and `no_ctg001250_curated.wrapped.fasta`? (please don't if it's from a human though) Or if it's too large, a relevant chunk of the .bam file that causes the same issue along with the corresponding .fasta.

**Comment by nmflack** [comment]
Thanks for your response Adam! The BAM is from a felid genome we're assembling, so I've attached the part of the modbam that aligns to the target region. Generated with:

`samtools view -b -F 4 pepper_modbam/mod.haplotagged.bam "ctg001660:139662493-139666498" > mini_aln/dmr_mod.bam`

I've also attached a Google Drive link to ctg01660 from `no_ctg001250_curated.wrapped.fasta`; the example doesn't work with just the DMR region. 

This code generates the same  "insufficient coverage" error, but leaving out the `--phased` flag results in a plot for both the full version and this minimal example:

```
methylartist locus \
--ref ctg001660.fa \
--motif CG \
-b dmr_mod.bam \
-i ctg001660:139662493-139666498
```

![mod haplotagged ctg001660_139662493_139666498 m ms1 smw24 locus meth](https://user-images.githubusercontent.com/109246157/182939003-79025347-3dad-466f-a223-2963c48e84b7.png)

Plots with phasing are successful for another version of the assembly haplotagged with Clair3, so I'm guessing the issue is related to PEPPER's output format.

Thank you,
Nicole

https://drive.google.com/drive/folders/1LyQcoXr4ofGDm1dMFCrHSAze0jteYMcv?usp=sharing

[methylartist_issue_21.zip](https://github.com/adamewing/methylartist/files/9263007/methylartist_issue_21.zip)

**Comment by adamewing** [comment]
The .bam you provided has "HP" tags but not "PS" tags and I had PS tags as an unnecessary requirement. Should work as of commit cf99013 and included in 1.2.4 (cat get via `pip`, conda takes some time to auto-update).

![dmr_mod ctg001660_139662493_139666498 m phased ms1 smw24 locus meth](https://user-images.githubusercontent.com/1037202/182984642-c3c2bc8d-67e5-42b8-9f96-a810e55af20b.png)

**Comment by nmflack** [comment]
Perfect, thank you! I re-haplotagged PEPPER's phased VCF output with longphase yesterday and got a plot; I checked and that version does have phase tags. I'll go ahead and update via `pip` to continue using the original PEPPER output. Much appreciated!
